test/e2e/storage/vsphere/vsphere_volume_datastore.go

```diff
@@ -63,21 +67,39 @@ var _ = utils.SIGDescribe("Volume Provisioning on Datastore [Feature:vsphere]",
 		}
 	})
 
-	It("verify dynamically provisioned pv using storageclass fails on an invalid datastore", func() {
+	It("should fail provisioning pv on invalid datastore", func() {
 		By("Invoking Test for invalid datastore")
 		scParameters[Datastore] = InvalidDatastore
 		scParameters[DiskFormat] = ThinDisk
-		err := invokeInvalidDatastoreTestNeg(client, namespace, scParameters)
+		err := invokeDatastoreTestNeg(client, namespace, scParameters)
 		Expect(err).To(HaveOccurred())
 		errorMsg := `Failed to provision volume with StorageClass \"` + DatastoreSCName + `\": The specified datastore ` + InvalidDatastore + ` is not a shared datastore across node VMs`
 		if !strings.Contains(err.Error(), errorMsg) {
 			Expect(err).NotTo(HaveOccurred(), errorMsg)
 		}
 	})
+
+	It("should fail provisioning pv on local datastore", func() {
+		By("Invoking Test for local datastore")
+		localDatastore = os.Getenv("LOCAL_DATASTORE")
```

Comment:
Can we use `GetAndExpectStringEnvVar` func here? 
https://github.com/kubernetes/kubernetes/blob/master/test/e2e/storage/vsphere/vsphere_common.go#L55

We should also move localDatastore to 
https://github.com/vmware/kubernetes/blob/master/test/e2e/storage/vsphere/vsphere_common.go#L25

Reply:
Above vsphere_common code was just merged 16 hours ago - it is not available in the current vmware/multi-vc-e2e branch. I will sync up the multi-vc-e2e branch with kubernetes/master to get those changes.

Reply:
I am wondering why do we need GetAndExpect here. What practice do we follow for our tests in case some env variable is not set- do we skip the test or do we fail it?

Reply:
@prashima we can fail or skip test depending on where the check is made. 

If check is made in `BeforeEach` and all `It` specs need this env variable to be set, we can skip running all tests, else we can also check this on individual bases in the 'It' spec and fail specific test.

Reply:
I don't think we should fail or skip the test depending on where the check is made. The test should always fail for production issue and skip for misconfiguration. I see a lot of existing tests fail for env variable not set, which is incorrect and we should fix that.

Reply:
@shaominchen suggested we should use `framework.skip` if environment is not set.
We may need to change - https://github.com/kubernetes/kubernetes/blob/master/test/e2e/storage/vsphere/vsphere_common.go#L64